src/Console/Command/FixCommand.php

```diff
@@ -263,6 +265,30 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         if (null !== $stdErr) {
             $stdErr->writeln(Application::getAboutWithRuntime(true));
+
+            if (version_compare(PHP_VERSION, ConfigInterface::PHP_VERSION_SYNTAX_SUPPORTED.'.99', '>')) {
+                $message = \sprintf(
+                    'PHP CS Fixer currently supports PHP syntax only up to PHP %s, current PHP version: %s. Use at your own risk.',
+                    ConfigInterface::PHP_VERSION_SYNTAX_SUPPORTED,
+                    PHP_VERSION
+                );
+
+                if (!$resolver->getUnsupportedPhpVersionAllowed()) {
+                    $message .= ' Add Config::setUnsupportedPhpVersionAllowed(true) to turn this into a warning.';
+                    $stdErr->writeln(\sprintf(
+                        $stdErr->isDecorated() ? '<bg=red;fg=white;>%s</>' : '%s',
+                        $message
+                    ));
+
+                    return 1;
+                }
+                $message .= ' Remove Config::setUnsupportedPhpVersionAllowed(true) to turn this into an error.';
+                $stdErr->writeln(\sprintf(
```

Comment:
Is it really necessary to show us the warning all the time? We would only ever see it if we had actively opted in to this behaviour. Personally, I don't see the benefit.

If you would't want to fully remove it, alternatively maybe this?

`--unsupported-php-version` and `Config::setUnsupportedPhpVersion` taking one of `disallowed|allowed|silently-allowed`, while `disallowed` is the default.

Reply:
You only see the warning if you're on an unsupported version of PHP which should hopefully be very rare since @keradus says the project should support new versions since day one.

Reply:
> The same behaviour as currently

Correct, as requested by @keradus in the review, my initial intent was to allow this by default. but he said it's too risky.

The major difference from this PR is you get to commit and ship your choice to opt-in to unsupported versions without passing weird env variables in 10 different ways so your whole team don't need to adjust how they use the tool just because you've updated PHP. This choice belongs in the config file, where it is right now.

Reply:
I am just talking about the message. The rest is clear. All good.